ui/src/Message.elm

```diff
@@ -76,6 +95,28 @@ update msg model =
                 Err error ->
                     ( { model | response = Error.handle error }, Cmd.none )
 
+        ThingMsg subMsg ->
+            let
+                ( updatedThing, thingCmd ) =
+                    Thing.update subMsg model.things token
+            in
+            ( { model | things = updatedThing }, Cmd.map ThingMsg thingCmd )
+
+        ChannelMsg subMsg ->
+            let
+                ( updatedChannel, channelCmd ) =
+                    Channel.update subMsg model.channels token
+            in
+            ( { model | channels = updatedChannel }, Cmd.map ChannelMsg channelCmd )
+
+        SelectedThing thingid ->
```

Comment:
Please move this message to `Thing.elm` module and add `.selectedThingId` to `Thing.Model`. Also, in `Thing.elm` module, use `retrieve` by importing it from `Channel` and move `Channel.RetrieveChannelsForThing` in `Thing.elm`.

Reply:
I did this but then code is too complicate, so I want to stay with my method of handling this.